src/ssl.c

```diff
@@ -8045,17 +8043,28 @@ WOLFSSL_EVP_PKEY* wolfSSL_d2i_PUBKEY(WOLFSSL_EVP_PKEY** out, unsigned char** in,
         long inSz)
 {
     WOLFSSL_EVP_PKEY* pkey = NULL;
+#if !defined(NO_RSA) ||   \
```

Comment:
This complex ifdef seems unnecessary. If the issue is "set, but not used" warning then down at the bottom before the "return" add `(void)mem; (void)memSz; (void)out;`. There is another case like this in `wolfSSL_d2i_PrivateKey_EVP`, which should also be fixed.

Reply:
@dgarske 

The changes are necessary to fix build error with